File: pj_backend/app/Models/Shipment.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Concerns\HasUuids;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasOne;

class Shipment extends Model
{
    use HasUuids;

    protected $fillable = [
        'customer_id',
        'driver_id',
        'origin',
        'destination',
        'transit_countries',
        'weight_kg',
        'size',
        'category_id',
        'vehicle_type_id',
        'price_breakdown',
        'total_price',
        'estimated_delivery_days',
        'status',
    ];

    protected $casts = [
        'transit_countries' => 'array',
        'price_breakdown' => 'array',
        'weight_kg' => 'float',
        'total_price' => 'float',
        'estimated_delivery_days' => 'integer',
        'customer_id' => 'integer',
        'category_id' => 'integer',
        'vehicle_type_id' => 'integer',
        'driver_id' => 'integer',
    ];

    public function customer(): BelongsTo
    {
        return $this->belongsTo(User::class, 'customer_id');
    }

    public function driver(): BelongsTo
    {
        return $this->belongsTo(User::class, 'driver_id');
    }

    public function category(): BelongsTo
    {
        return $this->belongsTo(Category::class);
    }

    public function vehicleType(): BelongsTo
    {
        return $this->belongsTo(VehicleType::class);
    }

    public function report(): HasOne
    {
        return $this->hasOne(Report::class);
    }
}

File: packages/pj_domain/lib/src/models/shipment.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'category.dart';
import 'vehicle_type.dart';

part 'shipment.freezed.dart';
part 'shipment.g.dart';

enum ShipmentStatus {
  pending,
  @JsonValue('in_transit')
  inTransit,
  delivered,
  reported,
}

@freezed
class Shipment with _$Shipment {
  const factory Shipment({
    required String id,
    @JsonKey(name: 'customer_id') required int customerId,
    @JsonKey(name: 'driver_id') int? driverId,
    required String origin,
    required String destination,
    @JsonKey(name: 'transit_countries') List<String>? transitCountries,
    @JsonKey(name: 'weight_kg') double? weightKg,
    @JsonKey(name: 'size') String? size,
    @JsonKey(name: 'category_id') required int categoryId,
    @JsonKey(name: 'vehicle_type_id') required int vehicleTypeId,
    @JsonKey(name: 'total_price') required double totalPrice,
    @JsonKey(name: 'estimated_delivery_days') required int estimatedDeliveryDays,
    @Default(ShipmentStatus.pending) ShipmentStatus status,
    Category? category,
    @JsonKey(name: 'vehicle_type') VehicleType? vehicleType,
    @JsonKey(name: 'price_breakdown') Map<String, dynamic>? priceBreakdown,
  }) = _Shipment;

  factory Shipment.fromJson(Map<String, dynamic> json) => _$ShipmentFromJson(json);
}
